java/internal/src/cuvs_java.c

```diff
@@ -296,20 +290,16 @@ void search_brute_force_index(cuvsBruteForceIndex_t index, float *queries, int t
   int64_t distances_shape[2] = {n_queries, topk};
   DLManagedTensor distances_tensor = prepare_tensor(distances, distances_shape, kDLFloat, 32, 2, kDLCUDA);
 
-  // unpack the incoming long into two 32bit ints
-  for (long i = 0; i < prefilter_data_length; i++) {
-    *(prefilter_data_32 + (2 * i)) = (int)(*(prefilter_data + i) >> 32);
-    *(prefilter_data_32 + ((2 * i) + 1)) = (int)*(prefilter_data + i);
-    //long l = (((long)*(prefilter_data_32 + (2 * i))) << 32) | (*(prefilter_data_32 + ((2 * i) + 1)) & 0xffffffffL);
-  }
-
   cuvsFilter prefilter;
   if (prefilter_data == NULL) {
     prefilter.type = NO_FILTER;
     prefilter.addr = (uintptr_t)NULL;
   } else {
-    int64_t prefilter_shape[1] = {(n_queries * n_rows + 31) / 32};
-    DLManagedTensor prefilter_tensor = prepare_tensor(prefilter_data_d, prefilter_shape, kDLUInt, 32, 1, kDLCUDA);
+    // Parse the filters data
+    int num_integers = (prefilter_data_length+63)/64 * 2;
+    int extraPaddingByteExists = prefilter_data_length % 64 > 32? 0: 1;
+    int64_t prefilter_shape[1] = {(prefilter_data_length + 31) / 32};
+    DLManagedTensor prefilter_tensor = prepare_tensor(prefilter_data, prefilter_shape, kDLUInt, 32, 1, kDLCUDA);
```

Comment:
I suppose the `DLManagedTensor prefilter_tensor` should be declared outside of `if` branch